pkg/crc/cache/cache.go

```diff
@@ -137,17 +165,20 @@ func (c *Cache) CheckVersion() error {
 	if c.version == "" {
 		return nil
 	}
-	currentVersion, err := c.getVersion()
+	currentVersion, err := c.getVersion(c)
```

Comment:
`c.getVersion(c)` this line seems particularly weird. 

I would expect getVersion to take a binary filepath as parameter and return a version. WDYT? `getVersionGeneric` only needs the binary. Making it static would also help.

Reply:
ah yes, should be better, thanks!